edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.Test/RuntimeInfoProviderTest.cs

```diff
@@ -332,5 +332,91 @@ public async Task GetModuleLogsTest()
             receivedLogs = Encoding.UTF8.GetString(buffer, 0, readBytes);
             Assert.Equal(dummyLogs, receivedLogs);
         }
+
+        [Fact]
+        [Unit]
+        public async Task GetModules_Handles_NotFoundException()
+        {
+            var missingContainerName = "container2";
+            var containersToReturn = new[] { "container1", "container2", "container3" };
+
+            var dockerClient = SetupDockerClient(missingContainerName, containersToReturn);
+            var provider = await RuntimeInfoProvider.CreateAsync(dockerClient);
+
+            var moduleInfo = await provider.GetModules(CancellationToken.None);
+
+            var actualContainerNames = moduleInfo
+                                            .Select(m => m.Name);
+
+            var expectedContainerNames = containersToReturn
+                                            .Where(c => !c.Equals(missingContainerName))
+                                            .Select(c => c.Substring(1))    // cuts the first letter
```

Comment:
why needs to cut the first letter?

Reply:
that is what RuntimeInfoProvider class does in the ExtractModuleRuntimeState() method. It would trim the module name, so I do the same in the test with the passed name strings. I suppose originally it trims a special character returned by docker api.

Reply:
I figured out container name returned from docker.DotNet NuGet will contains "/" prefix.  Therefore we remove 1st characters.

Therefore I think expectedContainersNames should be the same as containersToReturn; You need to update containerList variable in SetupDockerClient to returns each container name with "/" prefix.

Reply:
ok, now I am adding the prefix in the mocked docker api object, so the test method doesn't need to trim the first letter